WalletWasabi.Backend/Controllers/ChaumianCoinJoinController.cs

```diff
@@ -178,11 +178,27 @@ public async Task<IActionResult> PostInputsAsync([FromBody]InputsRequest request
 							{
 								return BadRequest("Provided input is neither confirmed, nor is from an unconfirmed coinjoin.");
 							}
-							// After 24 unconfirmed cj in the mempool dont't let unconfirmed coinjoin to be registered.
-							if (await Coordinator.IsUnconfirmedCoinJoinLimitReachedAsync())
+
+							// Check general limits: https://github.com/bitcoin/bitcoin/issues/14867#issuecomment-444113950
+							// These are not the only limits, but most of the limits are ensured by not letting non-CJ txs here.
+							// ToDo: explore all the limits to check so we can let in non-CJ unconfirmed transactions, too.
+							MempoolEntry entry = await RpcClient.GetMempoolEntryAsync(inputProof.Input.TransactionId);
+							if (entry.AncestorCount >= Helpers.Constants.DefaultAncestorLimit)
 							{
 								return BadRequest("Provided input is from an unconfirmed coinjoin, but the maximum number of unconfirmed coinjoins is reached.");
 							}
+							if (entry.AncestorVirtualSizeBytes >= ((Helpers.Constants.DefaultAncestorSizeLimitKb - 10) * 1000)) // Assume 10kb coinjoin.
+							{
+								return BadRequest("Provided input is from an unconfirmed coinjoin, but the maximum size of unconfirmed coinjoins is reached.");
+							}
+							if (entry.DescendantCount >= Helpers.Constants.DefaultDescendantLimit)
+							{
+								return BadRequest("Provided input is from an unconfirmed coinjoin, but the maximum number of unconfirmed descendants is reached.");
+							}
+							if (entry.DescendantVirtualSizeBytes >= ((Helpers.Constants.DefaultDescendantSizeLimitKb - 10) * 1000)) // Assume 10kb coinjoin.
+							{
+								return BadRequest("Provided input is from an unconfirmed coinjoin, but the maximum size of unconfirmed descendants is reached.");
+							}
```

Comment:
@lontivero Can you try to make sure somehow I didn't make an off by one error in any of the 4 limits?

Reply:
@nopara73 I am researching a bit on this because what I see in my test is that this RPC method `getmempoolentry` doesn't return the info that is used to check against the mempool policies.

Reply:
I've been pointed to this PR if that helps: https://github.com/bitcoin/bitcoin/pull/6771

Reply:
I hate to give bad news but the data returned by the rpc method is not the one needed to check the mempool acceptance policy. In other words, the comparisons you do (`if (entry.XXXX >= Helpers.Constants.DefaultYYYY`) make no sense.

In my PR for NBitcoin https://github.com/MetacoSA/NBitcoin/pull/611/files the unit test shows clearly that the method returns the counts and sizes of the **direct** ancestors and descendants. However, the mempool acceptance policy checks _if any ancestor would have 'n' or more in-mempool descendants_  and _if any ancestor would have more than 'n' kilobytes of in-mempool descendants_ 

After confirming this with the UT and reviewed the bitcoin project code and I can confirm that they check what they say they check and that the rpc method does not return the numbers that with need to compare.